File: Programacion de Inteligencia Artificial/Tema 1/Practica 1/codigoFuente/codigoFuente/PIA_S02_Files_Patterns/02_simple_patterns_error_control.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

int main (int argc, char *argv[]) {
  FILE *fp;
  char *line, *pattern;
  int occurrences;
  size_t len;
  ssize_t read;

  if (argc < 3) {
    fprintf(stderr, "Error! You need to provide the filepath and the pattern.\n");
    fprintf(stderr, "Sinopsis: ./02_simple_patterns_error_control <file> <pattern>\n");
    exit(EXIT_FAILURE);
  }
  
  pattern = argv[2];
  occurrences = 0;
  len = 0;

  if ((fp = fopen(argv[1], "r")) == NULL) {
    fprintf(stderr, "Error opening file %s!\n", argv[1]);
    exit(EXIT_FAILURE);
  }

  while ((read = getline(&line, &len, fp)) != -1) {
    if (strstr(line, pattern) != NULL) {
      occurrences++;
    }

  }

  fclose(fp);
  
  printf("The word %s appeared %d times in the file %s\n",
	 argv[2], occurrences, argv[1]);

  return 0;
}

File: Programacion de Inteligencia Artificial/1EV/Tema 1/Practica 1/codigoFuente/codigoFuente/PIA_S02_Files_Patterns/02_simple_patterns_error_control.py
#!/usr/bin/python3

# Import module sys
import sys


def main():
    # Variable to store how many times the pattern was found
    occurrences = 0

    try:
        # Open the file whose name is specified in the first argument
        with open(sys.argv[1]) as f:
            for line in f:
                # The words of every line are stored as a list
                words = line.split()
                # Check for matches
                for w in words:
                    # The pattern to be found is specified in sys.argv[2]
                    if w == sys.argv[2]:
                        occurrences += 1 

        print("The word", sys.argv[2], "appeared",
              occurrences, "times in the file", sys.argv[1])
        f.close()
    except IndexError:
        print("Error! You need to provide the filepath and the pattern.")
        print("Sinopsis: python3 02_simple_patterns_error_control <file> <pattern>")


if __name__ == "__main__":
    main()
